### PR TITLE
Vulnerability detector test module refactor: `test_providers_os`

### PR DESCRIPTION
**Pull request by fedepacher**
|Related issue|
|---|
|#2464|

## Description

This PR adds changes and enhancements to the testing framework for vulnerability detector.

The changes made are as follows:

### Added

- Add new functions to check log events.
- Add testing module: `test_provider_os` and data.

### Improvements

- Added a fast test (tier=0) and an extended test (tier=5).
- For tier=0 it checks that the expected logs appear in the `ossec.log` file.
- For tier=5 it checks that the expected logs appear in the `ossec.log` file and checks if the feeds of each provider are indexed in the respective table of the DB.

### Tests

- [x] `test_provider_os` - tier 0
- [x] `test_provider_os` - tier 5

### Outputs

<details>
<summary>test_provider_os - tier 0</summary>

```
======================= test session starts =======================
platform linux -- Python 3.6.8, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /mnt/qa/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: metadata-1.11.0, testinfra-5.0.0, html-3.1.1
collected 26 items                                                

test_vulnerability_detector/test_providers/test_providers_os.py . [  3%]
............sssssssssssss                                   [100%]

================= 13 passed, 13 skipped in 41.86s =================

```

</details>

<details>
<summary>test_provider_os - tier 5</summary>

```
======================= test session starts =======================
platform linux -- Python 3.6.8, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /mnt/qa/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: metadata-1.11.0, testinfra-5.0.0, html-3.1.1
collected 26 items                                                

test_vulnerability_detector/test_providers/test_providers_os.py s [  3%]
ssssssssssss.............                                   [100%]

=========== 13 passed, 13 skipped in 276.11s (0:04:36) ============

```

</details>

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py [hunk 1]
```diff
@@ -57,71 +57,40 @@
     - providers
 '''
 import os
-
 import pytest
+
+from datetime import date
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py:62]:
```suggestion
from datetime import date
```

File: tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py [hunk 2]
```diff
@@ -130,39 +99,111 @@ def test_providers(get_configuration, configure_environment, restart_modulesd):
...
+    input_description:
+        - The `test_os.yaml` file provides the module configuration for this test.
+
+    expected_output:
+        - r'Invalid option 'os' for '{provider_name}' provider.*'
+        - 'Starting '{provider_name}' database update'
+    '''
+    provider_name = metadata['provider_name']
+    operating_system = metadata['os']
+    if operating_system == '':
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py:134]:
```suggestion

    if operating_system == '':
```

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/test_os.yaml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    os: ''
+
+- name: 'Microsoft Security Update'
+  description: 'Microsoft Security Update provider'
+  configuration_parameters:
+    PROVIDER: 'msu'
+    OS: ''
+  metadata:
+    provider_name: 'Microsoft Security Update'
+    os: ''
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_providers/data/test_cases/test_os.yaml:116]: Where are the `Ubuntu focal`, `ArchLinux`, and `Debian Bullseye` providers?

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py (hunk 2 above)
Comment (fedepacher) [reply tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py:135]: Done in https://github.com/wazuh/wazuh-qa/pull/2549/commits/bcfe82b71c1952681de4c8862a2335609cd102b4

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**